desktop/core/src/desktop/api2_tests.py

```diff
@@ -699,6 +699,29 @@ def test_create(self):
         json.loads(Document2.objects.get(type='gist', uuid=gist2['uuid']).data)['statement_raw']
     )
 
+    # Create multiple gist directories and then make a new gist
+    home_dir = Directory.objects.get_home_directory(self.user)
+    gist_dir2 = Directory.objects.create(name='Gist', owner=self.user, parent_directory=home_dir)
```

Comment:
We should use the constant Document2.GIST_DIR for Gist, not the 'Gist' string to be consistent

Reply:
Also, how come we can have multiple Gist directory created in the first place?

We don't check for unique names? Recover from Trash?

Reply:
One way is creating them manually by using the '+' icon. This calls the [`create_directory`](https://github.com/cloudera/hue/blob/91b29ef3aaacdf8a6cacc9991e4dd6cb0192d7f8/desktop/core/src/desktop/api2.py#L514) which does not check for duplicated names
<img width="291" alt="Screenshot 2021-03-03 at 10 01 17 PM" src="https://user-images.githubusercontent.com/42064744/109838241-fd6c2500-7c6b-11eb-97a5-8b919378117e.png">

Will see if recovering from trash has the same issue or not

Reply:
Good point, and recovering from the Trash will have the same impact

Saving Directory does not any checks on duplicated named (as well as restore from Trash). Best for now would to just delete the dups when gettting back the gists (current approach)

```
class Document2(models.Model):
class Meta(object):
    unique_together = ('uuid', 'version', 'is_history')
```

```
class Directory(Document2):
  def save(self, *args, **kwargs):
    self.type = 'directory'
    super(Directory, self).save(*args, **kwargs)
```

Reply:
Sounds good!

Reply:
> We should use the constant Document2.GIST_DIR for Gist, not the 'Gist' string to be consistent

Shall we use this in [these lines](https://github.com/cloudera/hue/blob/b9e52f29050531e9d34c3b19ef6e13e14298a16f/desktop/core/src/desktop/api2_tests.py#L723-L727) as well? @romainr

Reply:
> Good point, and recovering from the Trash will have the same impact
> 
> Saving Directory does not any checks on duplicated named (as well as restore from Trash). Best for now would to just delete the dups when gettting back the gists (current approach)
> 
> ```
> class Document2(models.Model):
> class Meta(object):
>     unique_together = ('uuid', 'version', 'is_history')
> ```
> 
> ```
> class Directory(Document2):
>   def save(self, *args, **kwargs):
>     self.type = 'directory'
>     super(Directory, self).save(*args, **kwargs)
> ```

What if we dont allow making a duplicate directory in the first place (referring to create_directory here -> [potential fix](https://github.com/cloudera/hue/pull/1835))?
Will the suggested fix have any side effects??